indico/modules/events/api.py

```diff
@@ -538,6 +512,31 @@ def _update_query(self, query):
     def serialize_events(self, events):
         return map(self._build_event_api_data, events)
 
+    def _serialize_category_path(self, id):
+        category = Category.get(id)
+        visibility_title = None
+        path = [self._serialize_path_entry(category_data) for category_data in category.chain]
+        if not category.visibility:
+            visibility_title = 'Everywhere'
+        elif category.visibility == 0:
+            visibility_title = 'Nowhere'
+        else:
+            visibility_title = path[-category.visibility]['name']
```

Comment:
catch IndexError in case the visibility is higher than the length of the path (i'd return "Everywhere" in this case)